### PR TITLE
update market status fixer

### PR DESCRIPTION
**Pull request by GuillaumeDSM**
requires https://github.com/Drakkar-Software/OctoBot-Trading/pull/228 for CYTHON_IGNORE

### CODE REVIEW

File: octobot_trading/exchanges/util/exchange_market_status_fixer.pxd [hunk 1]
```diff
@@ -16,25 +16,26 @@
...
-    cdef object __calculate_amount(self)
-    cdef void __fix_market_status_limits_with_price(self)
-    cdef double __get_price_precision(self)
-    cdef void __fix_market_status_precision_with_price(self)
-    cdef void __fix_market_status_precision_with_specific(self)
-    cdef void __fix_market_status_limits_with_specific(self)
+    cpdef double _get_price_precision(self)
+    cpdef void _fix_market_status_precision_with_price(self)
+    cpdef void _fix_market_status_limits_with_specific(self)
+    cpdef void _fix_market_status_limits_with_price(self)
```
Comment (GuillaumeDSM) [inline octobot_trading/exchanges/util/exchange_market_status_fixer.pxd:36]: I had to make these cpdef to be able to call them in tests, what do you think about it @Herklos, should we keep cdef and avoid testing them directly ?

Comment (Herklos) [reply octobot_trading/exchanges/util/exchange_market_status_fixer.pxd:36]: We shouldn't change the cython header for test purposes. For similar cases I used to use CYTHON_IGNORE condition to pass the test when we are testing in compiled way.

File: octobot_trading/exchanges/util/exchange_market_status_fixer.pxd [hunk 2]
```diff
@@ -16,25 +16,26 @@
...
 cdef void calculate_costs(dict market_limit)
 cdef void calculate_prices(dict market_limit)
 cdef void calculate_amounts(dict market_limit)
 cdef object get_markets_limit(dict market_limit)
 cdef void fix_market_status_limits_from_current_data(dict market_limit)
-cdef bint check_market_status_limits(market_limit)
-cdef bint check_market_status_values(market_limit, bint zero_valid=*)
 
 cdef class ExchangeMarketStatusFixer:
     cdef public object market_status
```
Comment (Herklos) [inline octobot_trading/exchanges/util/exchange_market_status_fixer.pxd:29]: Should it be a dict?

File: octobot_trading/exchanges/util/exchange_market_status_fixer.pxd [hunk 3]
```diff
@@ -16,25 +16,26 @@
...
     cdef public object price_example
```
Comment (Herklos) [inline octobot_trading/exchanges/util/exchange_market_status_fixer.pxd:30]: I think it's a double

Comment (Herklos) [reply octobot_trading/exchanges/util/exchange_market_status_fixer.pxd:30]: And it shouldn't be public

File: octobot_trading/exchanges/util/exchange_market_status_fixer.pxd [hunk 4]
```diff
@@ -16,25 +16,26 @@
...
     cdef public object market_status_specific
```
Comment (Herklos) [inline octobot_trading/exchanges/util/exchange_market_status_fixer.pxd:31]: I think it's also a dict. I don't know if we are using it publicly

File: octobot_trading/exchanges/util/exchange_market_status_fixer.pxd (hunk 3 above)
Comment (Herklos) [reply octobot_trading/exchanges/util/exchange_market_status_fixer.pxd]: > I think it's a double

Don't forget the double :smile:

File: octobot_trading/exchanges/util/exchange_market_status_fixer.pxd (hunk 1 above)
Comment (Herklos) [reply octobot_trading/exchanges/util/exchange_market_status_fixer.pxd]: > We shouldn't change the cython header for test purposes. For similar cases I used to use CYTHON_IGNORE condition to pass the test when we are testing in compiled way.

To use CYTHON_IGNORE you need https://github.com/Drakkar-Software/OctoBot-Trading/pull/228

File: octobot_trading/exchanges/util/exchange_market_status_fixer.pxd [hunk 5]
```diff
@@ -16,25 +16,24 @@
...
 cpdef bint is_ms_valid(object value, bint zero_valid=*)
-
-cdef void calculate_costs(dict market_limit)
-cdef void calculate_prices(dict market_limit)
-cdef void calculate_amounts(dict market_limit)
-cdef object get_markets_limit(dict market_limit)
-cdef void fix_market_status_limits_from_current_data(dict market_limit)
-cdef bint check_market_status_limits(market_limit)
-cdef bint check_market_status_values(market_limit, bint zero_valid=*)
+cpdef bint check_market_status_limits(market_limit)
```
Comment (Herklos) [inline octobot_trading/exchanges/util/exchange_market_status_fixer.pxd:19]: Why do we need this functions to be public ?

Comment (GuillaumeDSM) [reply octobot_trading/exchanges/util/exchange_market_status_fixer.pxd:19]: every commit after tmp is a test, I'm trying to figure out why the compiled version is segfaulting

Comment (GuillaumeDSM) [reply octobot_trading/exchanges/util/exchange_market_status_fixer.pxd]: it looks like when making these private, python is enable to import them from the ExchangeMarketStatusFixer class

Comment (Herklos) [reply octobot_trading/exchanges/util/exchange_market_status_fixer.pxd]: :open_mouth:

Comment (GuillaumeDSM) [reply octobot_trading/exchanges/util/exchange_market_status_fixer.pxd]: further tests in progess

Comment (GuillaumeDSM) [reply octobot_trading/exchanges/util/exchange_market_status_fixer.pxd]: Issue fixed, a function was imported in another module

File: octobot_trading/exchanges/util/exchange_market_status_fixer.pxd (hunk 3 above)
Comment (GuillaumeDSM) [reply octobot_trading/exchanges/util/exchange_market_status_fixer.pxd]: With it being initialized bu default at None, it has to be object

**Pull request merged**